app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -66,7 +66,7 @@ def result
       service_provider: saml_request_service_provider,
       authn_context: requested_authn_contexts,
       authn_context_comparison: saml_request.requested_authn_context_comparison,
-      nameid_format: name_id_format,
+      nameid_format: saml_request.name_id_format,
```

Comment:
i'm not sure i understand why this is switching to the `saml_request` object's name_id_format -- can you walk me through the code flow a bit so i can understand better? thanks!

Reply:
As it is easier to reason about validation based on the original request, I was avoiding the name_id format transformation done by the `nameid_format` method and sending the originally requested `nameid_format` to the validator.